consensus/src/pipeline/header_processor/validate_in_isolation.rs

```diff
@@ -0,0 +1,53 @@
+use super::*;
+use crate::constants;
+use crate::errors::{ConsensusError, ConsensusResult, RuleError};
+use consensus_core::header::Header;
+use std::{
+    sync::Arc,
+    time::{SystemTime, UNIX_EPOCH},
+};
+
+impl HeaderProcessor {
+    pub(super) fn validate_header_in_isolation(self: &Arc<HeaderProcessor>, header: &Header) -> ConsensusResult<()> {
+        if header.hash == self.genesis_hash {
+            return Ok(());
+        }
+
+        self.check_header_version(header)?;
```

Comment:
I was expecting calls to `check_block_timestamp_in_isolation` and `check_parents_limit` here